File: Python/1279.py
ver = False
while True:
    try:
        n = int(input())
        if ver:
            print()
        b = False
        an = False
        
        if n % 400 == 0 or (n % 4 == 0 and n % 100 != 0):
            if n % 55 == 0:
                b = True
            print('This is leap year.')
            an = True
        if n % 15 == 0:
            print("This is huluculu festival year.")
            an = True
        if b:
            print('This is bulukulu festival year.')
            an = True
        
        if an == False:
            print("This is an ordinary year.")
        
        ver = True
    except EOFError:
        break

File: Java/1279.java
import java.math.BigInteger;
import java.util.Scanner;


public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int c = 1;
        while(sc.hasNextBigInteger()){
            BigInteger ano = sc.nextBigInteger();
            if(c > 1){
                System.out.println();
            }
            int ver = 0;
            if((ano.mod(BigInteger.valueOf(4)) == BigInteger.valueOf(0) && ano.mod(BigInteger.valueOf(100)) != BigInteger.valueOf(0)) || (ano.mod(BigInteger.valueOf(400)) == BigInteger.valueOf(0))){
                System.out.println("This is leap year.");
                ver = 1;
            }
            if(ano.mod(BigInteger.valueOf(15)) == BigInteger.valueOf(0)){
                System.out.println("This is huluculu festival year.");
                ver = 1;
            }
            if((ano.mod(BigInteger.valueOf(55)) == BigInteger.valueOf(0)) && ((ano.mod(BigInteger.valueOf(4)) == BigInteger.valueOf(0) && ano.mod(BigInteger.valueOf(100)) != BigInteger.valueOf(0)) || (ano.mod(BigInteger.valueOf(400)) == BigInteger.valueOf(0)))){
                System.out.println("This is bulukulu festival year.");
                ver = 1;
            }
            if(ver == 0){
                System.out.println("This is an ordinary year.");
            }
            c++;
        }

    }
}
